lighthouse-logger/index.js

```diff
@@ -128,6 +128,72 @@ class Log {
     marky.stop(id);
   }
 
+  /* eslint-disable no-invalid-this */
+  static timeDecorate(originalFn, opts) {
+    return function timeDecoratedFn(...args) {
+      const timeStartLogLevel = opts.timeStartLogLevel || 'log';
+      const timeEndLogLevel = opts.timeStartLogLevel || 'verbose';
+
+      let msg;
+      if (typeof opts.msg === 'string') {
+        msg = opts.msg;
+      } else if (typeof opts.msg === 'function') {
+        msg = opts.msg.bind(this)(...args);
+      }
+      if (!msg) {
+        throw new Error('expected msg');
+      }
+
+      let id;
+      if (typeof opts.id === 'string') {
+        id = opts.id;
+      } else if (typeof opts.id === 'function') {
+        id = opts.id.bind(this)(...args);
+      } else {
+        id = `lh:${originalFn.name}`;
+      }
+      if (!id) {
+        throw new Error('expected id');
+      }
+
+      const status = {msg, id};
+      Log.time(status, timeStartLogLevel);
+
+      let result;
+      try {
+        result = originalFn.bind(this)(...args);
+      } catch (err) {
+        err.stack = err.stack.replace(/.* at timeDecoratedFn .*\n/g, '');
+        throw err;
+      }
+
+      if (result && typeof result.then === 'function') {
```

Comment:
why not just add a oneline `if (typeof result.then === 'function') result = await result` in the try block above?

Reply:
oh right, silly me :) bummer!